app/models/rbac.rb

```diff
@@ -104,10 +104,8 @@ def self.rbac_class(scope)
     nil
   end
 
-  def self.rbac_instance(obj)
-    return obj                if apply_rbac_to_class?(obj.class)
-    return obj.resource       if obj.kind_of?(MetricRollup) || obj.kind_of?(Metric)
-    nil
+  def self.pluck_ids(targets)
+    targets.pluck(:id) if targets
```

Comment:
@chessbyte good question. I had to go deep to figure this out. I split out by the 3 methods we call:

**`get_self_service_objects`:**
no change: removed ordering, and bringing back just `id` instead of whole object with `id, type`. But `get_self_service_object_ids` just pulled out the `id`, so it is no real change. (Well, no longer creates an object for every Vm a user owns)

**`get_self_service_object_ids`:**
identical logic: `pluck(:id) if targets`

**`ids_via_descendants`:**
different: was `pluck(:id) unless targets.blank?` vs `pluck(:id) unless targets`
before: `targets.empty?` returns `nil`. after: `targets.empty?` returns `[]`

This is a wash when this result (`d_filtered_ids`) is actually used:
```ruby
filtered_ids += d_filtered_ids if d_filtered_ids.kind_of?(Array)
```
`nil` (skip the concatenation) vs `[]` (concatenate and change nothing) will have the same result

Benefit: 1 less `count(*)` database query.